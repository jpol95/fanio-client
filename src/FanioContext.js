import React from 'react'

const FanioContext = React.createContext({
    fandomList: [],
    reviewList: [],
    sectionList: [], 
    subList: [],
    tagList: [],
    reviewTagList: [],
    installmentList: [],
    handleSubmitReview: () => {}, 
    handleAddFandom: () => {}, 
    handleSubmitInstallments: () => {}, 
    handleGetReview: () => {}
})

export default FanioContext