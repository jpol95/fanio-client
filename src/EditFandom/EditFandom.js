import FandomForm from '../FandomForm/FandomForm'
import React from 'react'
import FetchService from '../FetchService'
import FanioContext from '../FanioContext'

export default class EditFandom extends React.Component{
    

  userId = Number(this.props.match.params.userId)
  fandomId = Number(this.props.match.params.fandomId)
  static contextType = FanioContext

  getFandomById(){
      console.log(this.context)
      return this.context.fandomList.find(fandom => fandom.id === this.fandomId)
  }
 
  handleSubmit = async (e, fandom) => {
    e.preventDefault()
    const fandomResult = await FetchService.patchFandom({...fandom, userId: this.userId}, this.fandomId)
    // console.log(fandom)
    this.props.history.push(`/users/${this.userId}/fandom-view/${fandomResult.id}`)
    this.context.handleEditFandom(fandomResult)
  }

  render(){
      if (!this.getFandomById()) return null
      const title = this.getFandomById().title
      console.log(title)
      const startInfo = {title}
      return ( 
      <>      
      <h3>Edit Fandom</h3>
      <FandomForm {...startInfo} handleSubmit={this.handleSubmit} />
      </>
      )
      //YOU ARE HERE. FIGURE OUT WHY THIS PROP IS NOT COMING THROUGH
  }
}