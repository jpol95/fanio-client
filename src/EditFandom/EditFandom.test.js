import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import EditFandom from './EditFandom';
import FanioContext from '../FanioContext'

describe('Edit fandom component', () => {
    const props = {
    match: {
        params: {
            userId: 1, 
            fandomId: 1
        }
    }, 
    installId: 1
}
const fandomList = [
    { id: 1, title: "Doctor Who", user: 1 }, //fk
    { id: 2, title: "Supernatural", user: 1 },
    { id: 3, title: "Parks and Recreation", user: 1 },
    { id: 4, title: "Buffy the Vampire Slayer", user: 1 },
    { id: 5, title: "Harry Potter", user: 1 },
  ];
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FanioContext.Provider value={{fandomList: fandomList}}> <EditFandom {...props} /> </FanioContext.Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<FanioContext.Provider value={{fandomList: fandomList}}> <EditFandom {...props} /> </FanioContext.Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});