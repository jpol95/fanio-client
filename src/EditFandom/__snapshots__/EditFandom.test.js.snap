// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Installment component renders the UI as expected 1`] = `
Array [
  " ",
  <h3>
    Edit Fandom
  </h3>,
  <div
    className="fandom"
  >
    <form
      className="create-fandom"
      onSubmit={[Function]}
    >
      <label
        htmlFor="workname"
      >
        Name of fandom*
      </label>
      <input
        defaultValue="Doctor Who"
        id="workname"
        onChange={[Function]}
        type="text"
      />
      <button
        className="submit-fandom-button"
        disabled={false}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
  " ",
]
`;
